### DISCUSSION TITLE
Why do I need to `@vue/babel-helper-vue-jsx-merge-props`  even if I don't use JSX

### DISCUSSION THREAD

**Discussion by vhoyer** [nuxt/nuxt#8722] [Questions (Nuxt 2)] [answered]
Like, I don't know why we need this package if I'm not using jsx, dispite that, this is the script with the largest evaluation time in my metrics.

![image](https://user-images.githubusercontent.com/11022464/106288808-41ac7580-6227-11eb-895d-94211f42ca8b.png)

Granted this app has a terrible performance overall, but here am I trying to make it better, and one of the things that seems helpful would be to remove this library, but I really don't know how, nor do I know why this is needed.

Anyone here can help me find the reason why this is needed in a nuxt project without jsx?

**Comment by danielroe** [comment]
@vhoyer This dependency is not bundled for the browser in a default Nuxt app, although Nuxt does support JSX out of the box. It is almost certainly something specific to your project. I would suggest trying to prepare a *minimal* reproduction (that is, just what is necessary to reproduce the problem). If you can do that, I'd be happy to take a look.

**Reply by vhoyer** [reply]
Oh, I see, ok, I will work on this, thanks for the help

**Comment by vhoyer** [comment, marked as answer]
Oh, I get it now, I kinda missed the fact that one of the dependencies where using, I was under the impression that we olny used pre-compiled stuff hence wouldn't need that library. Sorry for wasting your time here, thanks for the attention. :/

If anyone stumble into this page in the future, the way I checked where this was being used was pathetically simple in a way I'm kinda ashamed to have had opened this discussion here before checking.

1. build the project with `[nuxt.config.js]: build.analyze: true`,
2. use the built-in search of the bundle-analyzer to search for the package, in this case `babel-helper-vue-jsx-merge-props`;
3. then it lists all chunks where this dependency is being loaded, if you inspect what are the packages that are always packaged alongside the troublesome one, there is a high chance that this one was the culprit.

In my case it was the `egoist/vue-content-loader`.
